### ISSUE TITLE
external/generate_agent[BUG]

### ISSUE THREAD

**Issue by JeMunos** [BC-SECURITY/Empire#239]
**Empire Version**
- Empire 3.2.0 BC Security Fork

**OS Information (Linux flavor, Python version)**
- OS: Kali
- Python: 2.7.18

**Describe the bug**
Trying to create a basic external listener. Specifics provided below. 

**To Reproduce**

```
Description:
  Generates an agent code instance for a specified listener,
  pre-staged, and register the agent in the database. This
  allows the agent to begin beconing behavior immediately.

Options:

  Name     Required    Value                     Description
  ----     --------    -------                   -----------
  Listener True        http                      Listener to generate the agent for.
  Language True        powershell                Language to generate for the agent.
  OutFile  True        /tmp/agent                Output file to write the agent code to.

(Empire: external/generate_agent) > execute
[!] Exception: module 'string' has no attribute 'uppercase'
```

**Expected behavior**
```
(Empire: external/generate_agent) > set Listener http
(Empire: external/generate_agent) > set Language powershell
(Empire: external/generate_agent) > execute

[*] New agent N74NC8TD checked in
[+] Pre-generated agent 'N74NC8TD' now registered.
[*] powershell agent code for listener http with sessionID 'QKHQXGMU' written out to /tmp/agent
[*] Run sysinfo command after agent starts checking in!
```

**Screenshots**

**Additional context**

**Comment by Cx01N** [comment]
Sorry, the issue template is really picky on the formatting.
I am seeing the same issue you are which is an easy fix, but there are some other agent issues that haven't been updated for a long time. Let me see if I can get it working and put out a branch for you to test.

**Comment by Cx01N** [comment]
It seems to be working on my end. Let me know if the [branch](https://github.com/BC-SECURITY/Empire/tree/external_agent_fix) works for you and I will get it merged for the next release.

**Comment by JeMunos** [comment]
So i tried the fix "chars = string.ascii_uppercase + string.digits"
However this results in another bug that I don't know how to track down. 
```
(Empire: external/generate_agent) > info

              Name: Generate Agent
            Module: external/generate_agent
        NeedsAdmin: False
         OpsecSafe: True
          Language: Python
        Background: True
   OutputExtension: None

Authors:
  @harmj0y

Description:
  Generates an agent code instance for a specified listener,
  pre-staged, and register the agent in the database. This
  allows the agent to begin beconing behavior immediately.

Options:

  Name     Required    Value                     Description
  ----     --------    -------                   -----------
  Listener True        meterpreter               Listener to generate the agent for.
  Language True        powershell                Language to generate for the agent.
  OutFile  True        /tmp/agent                Output file to write the agent code to.

(Empire: external/generate_agent) > execute
[!] Exception: 'StagingKey'
```

**Comment by Cx01N** [comment]
I just noticed that you listed Python: 2.7.18. You'll have to make sure you are running Python 3.7 or 3.8. We no longer support Python 2.

**Comment by JeMunos** [comment]
Fair. Stupid question. 
How do I force it to run using the installed version of python3? Kali comes with both

**Comment by JeMunos** [comment]
So I used this guide: https://linuxconfig.org/how-to-change-default-python-version-on-debian-9-stretch-linux
and set my default python version to 3.8

```
ec2-user@kali:~$ python --version
Python 3.8.3
ec2-user@kali:~$
```

However I'm still seeing the same issue with the staging key error I posted above. Could this be because the initial setup I did was using 2.7 and I need to reset the environment? Maybe. I'll try that.
